src/main/java/com/odysseusinc/arachne/portal/model/DataNode.java

```diff
@@ -91,15 +91,25 @@
     @OneToMany(mappedBy = "dataNode", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
     private Set<DataNodeUser> dataNodeUsers = new HashSet<>();
 
-    @Column(name = "atlas_version")
-    private String atlasVersion;
+    @OneToMany(mappedBy = "dataNode", fetch = FetchType.LAZY)
+    private Set<Atlas> atlasList;
 
     @Transient
     private Set<ArachnePermission> permissions;
 
     @Column
     private Boolean published;
 
+    public String getAtlasVersion() {
+
+        return this.getAtlasList().stream()
+                .filter(a -> a.getVersion() != null)
```

Comment:
can be replaced by Objects.nonNull

Reply:
It already contains "Objects" from Google package, so...